happi/cli.py

```diff
@@ -1075,6 +1077,73 @@ def audit(
         click.echo(f'# devices failed: {len(unique_fails)} / {len(results)}')
 
 
+@happi_cli.command()
+@click.pass_context
+@click.option('--fix-optional/--ignore-optinal', 'fix_optional', default=False,
+              help='Also prompt for user input on optional information')
+@click.option('--glob/--regex', 'use_glob', default=True,
+              help='Use glob (default) or regex style search terms. '
+              'Only relevant if search_criteria are provided.')
+@click.argument('search_criteria', nargs=-1)
+def repair(
+    ctx,
+    fix_optional: bool,
+    use_glob: bool,
+    search_criteria: tuple[str, ...]
+):
+    """
+    Repair the database.
+
+    Repairs all entries matching SEARCH_CRITERIA, repairs entire database otherwise.
+    """
+    logger.debug('starting repair block')
+
+    client: happi.Client = ctx.obj
+    if search_criteria:
+        results = search_parser(client, use_glob, search_criteria)
+    else:
+        # run repair on all items
+        results = search_parser(client, True, '*')
+
+    for res in results:
+        # fix mandatory info with missing defaults
+        req_info = (info for info in find_unfilled_mandatory_info(res))
+
+        if fix_optional:
+            empty_opt = find_unfilled_optional_info(res)
+            req_info = itertools.chain(req_info, empty_opt)
+
+        try:
+            req_field = next(req_info)
+        except StopIteration:
+            req_field = False
+
+        res_name = res['_id']
+
+        # fix each mandatory field
+        logger.info(f'repairing ({res_name})...')
+        while req_field:
```

Comment:
Why the fancy iterator usage?
Would `for req_field in req_info` not suffice?

Reply:
Ah this was necessary before I remembered about the `prompt_for_entry` helper.  Simple will do here